datafusion/core/src/physical_planner.rs

```diff
@@ -1941,6 +2141,59 @@ impl DefaultPhysicalPlanner {
         let mem_exec = MemoryExec::try_new(&partitions, schema, projection)?;
         Ok(Arc::new(mem_exec))
     }
+
+    fn create_project_physical_exec(
+        &self,
+        session_state: &SessionState,
+        input_exec: Arc<dyn ExecutionPlan>,
+        input: &Arc<LogicalPlan>,
+        expr: &[Expr],
+    ) -> Result<Arc<dyn ExecutionPlan>> {
```

Comment:
Just extracted this into separate function as it was also used by `Join` to create the physical projections that are added during this planning if join has expression equijoin keys.